x-pack/test/functional/apps/ml/data_visualizer/index_data_visualizer_grid_in_dashboard.ts

```diff
@@ -0,0 +1,160 @@
+import { farequoteLuceneFiltersSearchTestData } from './index_test_data';
+
+const SHOW_FIELD_STATISTICS = 'discover:showFieldStatistics';
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const PageObjects = getPageObjects(['common', 'discover', 'timePicker', 'settings', 'dashboard']);
+  const ml = getService('ml');
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const toasts = getService('toasts');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+
+  const selectIndexPattern = async (indexPattern: string) => {
+    await retry.tryForTime(2 * 1000, async () => {
+      await PageObjects.discover.selectIndexPattern(indexPattern);
+      const indexPatternTitle = await testSubjects.getVisibleText('indexPattern-switch-link');
+      expect(indexPatternTitle).to.be(indexPattern);
+    });
+  };
+
+  const clearAdvancedSetting = async (propertyName: string) => {
+    await retry.tryForTime(2 * 1000, async () => {
+      await PageObjects.common.navigateToUrl('management', 'kibana/settings', {
+        shouldUseHashForSubUrl: false,
+      });
+      if ((await PageObjects.settings.getAdvancedSettingCheckbox(propertyName)) === 'true') {
+        await PageObjects.settings.clearAdvancedSettings(propertyName);
+      }
+    });
+  };
+
+  const setAdvancedSettingCheckbox = async (propertyName: string, checkedState: boolean) => {
+    await retry.tryForTime(2 * 1000, async () => {
+      await PageObjects.common.navigateToUrl('management', 'kibana/settings', {
+        shouldUseHashForSubUrl: false,
+      });
+      await testSubjects.click('settings');
+      await toasts.dismissAllToasts();
+      await PageObjects.settings.toggleAdvancedSettingCheckbox(propertyName, checkedState);
+    });
+  };
+
+  function runTests(testData: TestData) {
+    const savedSearchTitle = `Field stats for ${testData.suiteTitle} ${Date.now()}`;
+    const dashboardTitle = `Dashboard for ${testData.suiteTitle} ${Date.now()}`;
+
+    describe(`with ${testData.suiteTitle}`, function () {
+      after(async function () {
+        await ml.testResources.deleteSavedSearchByTitle(dashboardTitle);
+        await ml.testResources.deleteDashboardByTitle(dashboardTitle);
+      });
+
+      it(`saves search with Field statistics table in Discover`, async function () {
+        await setAdvancedSettingCheckbox(SHOW_FIELD_STATISTICS, true);
+
+        await PageObjects.common.navigateToApp('discover');
+        if (testData.isSavedSearch) {
+          await retry.tryForTime(2 * 1000, async () => {
+            await PageObjects.discover.loadSavedSearch(testData.sourceIndexOrSavedSearch);
+          });
+        } else {
+          await selectIndexPattern(testData.sourceIndexOrSavedSearch);
+        }
+        await PageObjects.timePicker.setAbsoluteRange(
+          'Jan 1, 2016 @ 00:00:00.000',
+          'Nov 1, 2020 @ 00:00:00.000'
```

Comment:
We're using this time range three times in this file, might be worth a const?

Reply:
Updated here https://github.com/elastic/kibana/pull/116774/commits/b9a99bd5223d12dc354a34f48feb0fde9fb5e773